### PR TITLE
Added more compiler warnings

### LINKED ISSUE #18: Add more compiler warnings and fixed discovered issues
Related: https://github.com/wordpress-mobile/WordPressCom-Stats-iOS/issues/218

### PR DESCRIPTION
**Pull request by astralbodies**
Closes #18 

Added a few more compiler warnings to the configuration and fixed some issues discovered with compilation and static analysis.

Needs Review: @aerych

### CODE REVIEW

File: Automattic-Tracks-iOS/TracksEvent.m [hunk 1]
```diff
@@ -206,7 +206,7 @@ - (BOOL)validateDeviceProperties:(id *)ioValue error:(NSError * __autoreleasing
 
 - (BOOL)validateUserProperties:(id *)ioValue error:(NSError * __autoreleasing *)outError
 {
-    if (ioValue == nil || ([(NSDictionary *)*ioValue count] > 0)) {
+    if (ioValue != nil && ([(NSDictionary *)*ioValue count] > 0)) {
```
Comment (aerych) [inline Automattic-Tracks-iOS/TracksEvent.m]: I think you could omit checking `ioValue != nil` in these three conditionals. Anytime `ioValue == nil` the `[ioValue count] > 0` check will return false.

**Comment by aerych** [comment]
One tiny comment, otherwise :shipit: :)

File: Automattic-Tracks-iOS/TracksEvent.m (hunk 1 above)
Comment (astralbodies) [inline Automattic-Tracks-iOS/TracksEvent.m]: True dat. Going to remove those checks!

**Comment by astralbodies** [comment]
Thanks again @aerych!

**Pull request merged**